code/modules/pool/pool_main.dm

```diff
@@ -0,0 +1,187 @@
+/turf/open/MouseDrop_T(atom/from, mob/user)
+	// I could make this /open/floor and not have the !istype but ehh - kev
+	if(isliving(from) && SEND_SIGNAL(from, COMSIG_IS_SWIMMING) && isliving(user) && ((user == from) || user.CanReach(from)) && !user.IsStun() && !user.IsKnockdown() && !user.incapacitated() && !istype(src, /turf/open/pool))
+		var/mob/living/L = from
+		//The element only exists if you're on water and a living mob, so let's skip those checks.
+		var/pre_msg
+		var/post_msg
+		if(user == from)
+			pre_msg = "<span class='notice'>[L] is getting out of the pool.</span>"
+			post_msg = "<span class='notice'>[L] gets out of the pool.</span>"
+		else
+			pre_msg = "<span class='notice'>[L] is being pulled out of the pool by [user].</span>"
+			post_msg = "<span class='notice'>[user] pulls [L] out of the pool.</span>"
+		L.visible_message(pre_msg)
+		if(do_mob(user, L, 20))
+			L.visible_message(post_msg)
+			L.forceMove(src)
+	else
+		return ..()
+
+// Exit check
+/turf/open/pool/Exit(atom/movable/AM, atom/newloc)
+	if(!AM.has_gravity(src))
+		return ..()
+	if(isliving(AM) || istype(AM, /obj/structure))
+		if(AM.throwing)
+			return ..()			//WHEEEEEEEEEEE
+		if(istype(AM, /obj/structure) && isliving(AM.pulledby))
+			return ..()			//people pulling stuff out of pool
+		if(!ishuman(AM))
+			return ..()			//human weak, monkey (and anyone else) ook ook eek eek strong
+		return istype(newloc, type)
+	return ..()
+
+// Exited logic
+/turf/open/pool/Exited(atom/A, atom/newLoc)
+	. = ..()
+	if(isliving(A))
+		var/turf/open/pool/P = newLoc
+		if(!istype(P) || (P.controller != controller))
+			controller?.mobs_in_pool -= A
+
+// Entered logic
+/turf/open/pool/Entered(atom/movable/AM, atom/oldloc)
+	if(istype(AM, /obj/effect/decal/cleanable))
+		var/obj/effect/decal/cleanable/C = AM
+		if(prob(C.bloodiness))
+			controller.set_bloody(TRUE)
+		QDEL_IN(AM, 25)
+		animate(AM, alpha = 10, time = 20)
+		return ..()
+	if(!AM.has_gravity(src))
+		return ..()
+	if(isliving(AM))
+		var/mob/living/victim = AM
+		if(!SEND_SIGNAL(victim, COMSIG_IS_SWIMMING))		//poor guy not swimming time to dunk them!
+			victim.AddElement(/datum/element/swimming)
+			controller.mobs_in_pool += victim
+			if(locate(/obj/structure/pool/ladder) in victim.loc)		//safe climbing
+				return
+			if(iscarbon(AM))		//FUN TIME!
+				var/mob/living/carbon/H = victim
+				if(filled)
+					if (H.wear_mask && H.wear_mask.flags_cover & MASKCOVERSMOUTH)
+						H.visible_message("<span class='danger'>[H] falls in the water!</span>",
+											"<span class='userdanger'>You fall in the water!</span>")
+						playsound(src, 'sound/effects/splash.ogg', 60, TRUE, 1)
+						H.Knockdown(20)
+						return
+					else
+						H.Knockdown(60)
+						H.adjustOxyLoss(5)
+						H.emote("cough")
+						H.visible_message("<span class='danger'>[H] falls in and takes a drink!</span>",
+											"<span class='userdanger'>You fall in and swallow some water!</span>")
```

Comment:
based on this message, it seems like you would take a gulp of whatever reagents in the pool, but this is not the case. Consider adding that?